library/src/blas3/rocblas_trmm.cpp

```diff
@@ -1,395 +1,160 @@
-
-    @param[in]
-    n       rocblas_int.
-            n specifies the number of columns of B. n >= 0.
-
-    @param[in]
-    alpha
-            alpha specifies the scalar alpha. When alpha is
-            zero then A is not referenced and B need not be set before
-            entry.
-
-    @param[in]
-    A       pointer storing matrix A on the GPU.
-            of dimension ( lda, k ), where k is m
-            when  rocblas_side_left  and
-            is  n  when  rocblas_side_right
-            only the upper/lower triangular part is accessed.
-
-    @param[in]
-    lda     rocblas_int.
-            lda specifies the first dimension of A.
-            if side = rocblas_side_left,  lda >= max( 1, m ),
-            if side = rocblas_side_right, lda >= max( 1, n ).
-
-    @param[in]
-    B       pointer storing matrix B on the GPU.
-
-    @param[in]
-    ldb    rocblas_int.
-           ldb specifies the first dimension of B. ldb >= max( 1, m ).
-
-    @param[in,output]
-    C       pointer storing matrix C on the GPU.
-
-    @param[in]
-    ldc    rocblas_int.
-           ldb specifies the first dimension of C. ldc >= max( 1, m ).
-
-    ********************************************************************/
-
     template <typename T>
-    rocblas_status rocblas_trmm(rocblas_handle    handle,
-                                rocblas_side      side,
-                                rocblas_fill      uplo,
-                                rocblas_operation transA,
-                                rocblas_diagonal  diag,
-                                rocblas_int       M,
-                                rocblas_int       N,
-                                const T*          alpha,
-                                const T*          A,
-                                rocblas_int       lda,
-                                const T*          B,
-                                rocblas_int       ldb,
-                                T*                C,
-                                rocblas_int       ldc)
+    rocblas_status rocblas_trmm_impl(rocblas_handle    handle,
+                                     rocblas_side      side,
+                                     rocblas_fill      uplo,
+                                     rocblas_operation transa,
+                                     rocblas_diagonal  diag,
+                                     rocblas_int       m,
+                                     rocblas_int       n,
+                                     const T*          alpha,
+                                     const T*          a,
+                                     rocblas_int       lda,
+                                     T*                c,
+                                     rocblas_int       ldc)
     {
         if(!handle)
             return rocblas_status_invalid_handle;
-        if(!alpha)
-            return rocblas_status_invalid_pointer;
 
         auto layer_mode = handle->layer_mode;
```

Comment:
We disable logging when `handle->is_device_memory_size_query()`.

Reply:
Disabled logging when is_device_memory_size_query